.github/workflows/test-plugins.yml

```diff
@@ -0,0 +1,40 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow
+
+name: Test Plugins
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: ${{ matrix.plugin }}
+    strategy:
+      matrix:
+        plugin:
+          - 'user-switching'
+      fail-fast: false
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          repository: johnbillion/${{ matrix.plugin }}
+          token: ${{ secrets.PLUGINS_PUSH_TOKEN }}
+
+      - name: Update tests reference
+        run: |
+          sed -i 's/reusable-integration-tests.yml@trunk/reusable-integration-tests.yml@${{ github.head_ref }}/g' .github/workflows/integration-tests.yml
+
+      - name: Open PR
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.PLUGINS_PUSH_TOKEN }}
```

Comment:
There is no need for an extra token. `github.token` is enough.
It needs `permissions:` for PR-s.

Reply:
My understanding is that the default token is scoped to the current repo. This action is pushing to and opening a PR against another repo.

Reply:
Yeah looks like you can't use the default token: https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs